src/Microsoft.AspNetCore.Razor.Language/RazorCodeDocumentExtensions.cs

```diff
@@ -29,6 +29,26 @@ public static void SetTagHelperContext(this RazorCodeDocument document, TagHelpe
             document.Items[typeof(TagHelperDocumentContext)] = context;
         }
 
+        public static IReadOnlyList<TagHelperDescriptor> GetTagHelpers(this RazorCodeDocument document)
+        {
+            if (document == null)
+            {
+                throw new ArgumentNullException(nameof(document));
+            }
+
+            return (document.Items[typeof(TagHelpersHolder)] as TagHelpersHolder)?.TagHelpers;
+        }
+
+        public static void SetTagHelpers(this RazorCodeDocument document, IReadOnlyList<TagHelperDescriptor> tagHelpers)
```

Comment:
Lets mark both of these as internal. There could be some confusion from an end users perspective about what these methods do (do they calculate TagHelpers? how is it different from the taghelper feature? etc.)